cotainr/tests/container/test_singularity_sandbox.py

```diff
@@ -83,20 +85,70 @@ def test_add_twice(
+            0o007,  # 660 (rw-rw----)   770 (rwxrwx---)  # umask used on LUMI
+            0o022,  # 644 (rw-r--r--)   755 (rwxr-xr-x)  # default umask in most cases
+            0o027,  # 640 (rw-r-----)   750 (rwxr-x---)
+            0o077,  # 600 (rw-------)   700 (rwx------)
+        ],
+    )
+    def test_correct_umask(
+        self,
+        umask,
+        context_set_umask,
+        data_cached_alpine_sif,
+        singularity_exec,
+        tmp_path,
+    ):
+        with context_set_umask(umask):  # default umask on LUMI
+            tmp_path.chmod(
+                # apply effective umask permissions to tmp_path as well
+                # it is created without the custom umask
+                0o777 - umask
+            )
+            built_image_path = tmp_path / "built_image_6021"
+            with SingularitySandbox(base_image=data_cached_alpine_sif) as sandbox:
+                # Test file permissions
+                env_file = sandbox.sandbox_dir / ".singularity.d/env/92-cotainr-env.sh"
+                sandbox._create_file(env_file=env_file)
+                assert env_file.exists()
+                test_file_mode = env_file.stat().st_mode
+                # file permissions extracted from the last 3 octal digits of st_mode
+                test_file_permissions = test_file_mode & 0o777
+                assert oct(test_file_permissions) == "0o644"
+
+                # Test source 92-cotainr-env.sh
+                sandbox.add_to_env(
+                    shell_script="echo 'we can read the env file, 6021!'"
+                )
+                sandbox.build_image(path=built_image_path)
+
+            assert singularity_exec(
+                f"{built_image_path} echo 'more text...'"
+            ).stdout == ("we can read the env file, 6021!\nmore text...\n")
+
     def test_newline_encapsulation(
-        self, patch_disable_stream_subprocess, patch_fake_singularity_sandbox_env_folder
+        self,
+        patch_disable_stream_subprocess,
+        patch_fake_singularity_sandbox_env_folder,
     ):
         with SingularitySandbox(base_image="my_base_image_6021") as sandbox:
             env_file = sandbox.sandbox_dir / ".singularity.d/env/92-cotainr-env.sh"
             shell_script = "fancy shell_script\nas a double line string"
             sandbox.add_to_env(shell_script=shell_script)
             assert env_file.read_text() == shell_script + "\n"
 
-    def test_shell_script_append(
-        self, patch_disable_stream_subprocess, patch_fake_singularity_sandbox_env_folder
-    ):
-        with SingularitySandbox(base_image="my_base_image_6021") as sandbox:
+    @pytest.mark.singularity_integration
+    def test_existing_file(self, data_cached_alpine_sif):
+        with SingularitySandbox(base_image=data_cached_alpine_sif) as sandbox:
             env_file = sandbox.sandbox_dir / ".singularity.d/env/92-cotainr-env.sh"
+            env_file.touch()  # Note, file permissions corresponds to system default
```

Comment:
I don't touching the file is necessary - ` env_file.write_text( )` should create it, if it doesn't exist.
The comment is relevant, though, so maybe just move it to the `write_text( )` line.